string clear_area
string block_area
